### ISSUE TITLE
Search and queue/play media

### ISSUE THREAD

**Issue by jonkristian** [jonkristian/casatunes#5]
Would be nice to be able to search media and queue or play directly.
- Search for music in current collection `GET: /media/search/{mcId}/searchText`
- Search for music via Home Assistant [Assist](https://www.home-assistant.io/docs/assist/) would be a goal. 

CasaTunes already supports search in their api, we might be limited by Home Assistant here but I've seen some implementation code for this on the [Mopidy](https://github.com/bushvin/hass-integrations/tree/main/custom_components/mopidy) component.

**Comment by jonkristian** [comment]
Now that Home Assistant 2023.8 has come out with [wildcard support for sentence triggers](https://www.home-assistant.io/blog/2023/08/02/release-20238/#wildcard-support-for-sentence-triggers) I was inspired to start working on this issue and eventually get this working with assist too.

Working code in master, but needs testing.

**Comment by alphasixtyfive** [comment]
I think it would be nice to also expose doorbell, chime and TTS as services.

**Comment by jonkristian** [comment]
@alphasixtyfive I'm currently looking into TTS service.

Have you ever tried the SSML part of TTS? For me, it seems to error out. If you got it working it would be helpful to see how your request is built.

```
{
  "ResponseStatus": {
    "ErrorCode": "system/tts/input/Here are <say-as interpret-as=\"characters\">SSML</say-as> samples./zones/0 was not found",
    "Message": "system/tts/input/Here are <say-as interpret-as=\"characters\">SSML</say-as> samples./zones/0 was not found",
    "StackTrace": "[Fallback: 8/9/2023 7:17:26 PM]:\n[REQUEST: {Path:\"system/tts/input/Here are <say-as interpret-as=\"\"characters\"\">SSML</say-as> samples./zones/0\"}]\nServiceStack.Common.Web.HttpError: system/tts/input/Here are <say-as interpret-as=\"characters\">SSML</say-as> samples./zones/0 was not found\r\n   at CasaTunes.REST.ServiceInterface.CasaService.Any(Fallback request)\r\n   at lambda_method(Closure , Object , Object )\r\n   at ServiceStack.ServiceHost.ServiceRunner`1.Execute(IRequestContext requestContext, Object instance, TRequest request)",
    "Errors": []
  }
}
```

Just tested with a couple of examples from [google ssml docs](https://cloud.google.com/text-to-speech/docs/ssml).

**Comment by alphasixtyfive** [comment]
I have not tried this I'm afraid.

**Comment by jonkristian** [comment]
Ok. New code in master. Improvements to the search and services for tts and doorbell. I could not get SSML to work, but other than that it seems to work pretty well. I think the SSML might be either an upstream bug or I am doing it waay wrong :dagger:

**Comment by alphasixtyfive** [comment]
I think you might want to rename doorbell to chime as what the code does not is call chime endpoint. Doorbell simply plays defined chime in seleted zones but this is a slightly different thing.

**Comment by jonkristian** [comment]
As I understand it, both endpoints starts with `/system/doorbell`, but one is for the whole house and one is for specific zone(s). I was actually thinking about the following: When zone(s) are chosen we can use `/system/doorbell/zones/zone_id/chimes/chime` but if no specific zone; we call `/system/doorbell`. But what do you think, would a separate service for doorbell be better? For me, it doesnt really matter, but it would keep the number of services low, just that the logic is inside a service, if possible of course.

**Comment by alphasixtyfive** [comment]
I fully agree with the logic. I was also thinking of perhaps having `preWait` and `postWait` as default options configurable for the component?
